drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -116,217 +121,194 @@ impl Handler {
         }
 
         // Check for the minimum receive length being satisfied.
-        if rx_buf.len() < MIN_MSG_SIZE {
-            tx_payload[0] = MsgError::BadMessageLength as u8;
-            return compose(MsgType::ErrorRsp, 1, tx_buf).ok();
+        if rx_bytes < MIN_MSG_SIZE {
+            return Some(tx_buf.error_rsp(SprotError::BadMessageLength));
         }
 
         // Parse the header which also checks the CRC.
-        let (msgtype, rx_payload) = match parse(rx_buf) {
-            Ok((msgtype, payload)) => {
-                self.prev = PrevMsg::Good(msgtype);
-                (msgtype, payload)
+        let rxmsg = match rx_buf.parse_header(rx_bytes) {
+            Ok(header) => {
+                // We want to ensure the number of bytes received matches the header's
+                // expected payload size.
+                let expected_payload = rx_bytes - MIN_MSG_SIZE;
+                if header.payload_len as usize != rx_bytes - MIN_MSG_SIZE {
```

Comment:
```suggestion
                if header.payload_len as usize != expected_payload {
```

Reply:
Fixed in [f86812d](https://github.com/oxidecomputer/hubris/pull/965/commits/f86812d56e41435275dab8b4cf32982f88d4c35f)